### PR TITLE
[AMLII-2060] Add more robust error handling for files

### PR DESCRIPTION
**Pull request by soberpeach**
### What does this PR do?

Adds more robust error handling for files where settings are improperly set or read incorrectly.

### Motivation

Previous changes induced a panic in the agent when `integrations_logs_disk_ratio` was read or set improperly, this change mitigates that.

### Describe how to test/QA your changes

Since this is a fix, just maake sure the agent runs correctly and all tests pass.

### Possible Drawbacks / Trade-offs

N/A

### CODE REVIEW

File: pkg/logs/launchers/integration/launcher_test.go [hunk 1]
```diff
@@ -156,6 +156,14 @@ func (suite *LauncherTestSuite) TestDeleteFile() {
 	assert.True(suite.T(), os.IsNotExist(err))
 }
 
+// TestNegativeCombinedUsageMax ensures errors in combinedUsageMax don't result
+// in panics from `deleteFile`
+func (suite *LauncherTestSuite) TestNegativeCombinedUsageMax() {
+	suite.s.combinedUsageMax = -1
```
Comment (gh123man) [inline pkg/logs/launchers/integration/launcher_test.go:162]: Maybe we can add tests that cover the behavior when: 

`combinedUsageMax` is `0`, `1`, `100` (something large enough to store one or two logs). 
In each case, write several logs from several sources and assert that everything behaves as expected. 
Just to put the components through it's paces and make sure it behaves correctly in anomalous cases.

File: pkg/logs/launchers/integration/launcher.go [hunk 2]
```diff
@@ -181,6 +196,10 @@ func (s *Launcher) receiveLogs(log integrations.IntegrationLog) {
 	// deleting files until total usage falls below the set maximum
 	for s.combinedUsageSize+logSize > s.combinedUsageMax {
 		leastRecentlyModifiedFile := s.getLeastRecentlyModifiedFile()
+		if leastRecentlyModifiedFile == nil {
+			ddLog.Warn("getLeastRecentlyModifiedFile returned nil, skipping writing log to file.")
```
[the remark this answers is not in the document]
Comment (soberpeach) [reply pkg/logs/launchers/integration/launcher.go:200]: Makes sense 👍

File: pkg/logs/launchers/integration/launcher.go [hunk 3]
```diff
@@ -363,6 +386,11 @@ func (s *Launcher) scanInitialFiles(dir string) error {
 	for s.combinedUsageSize > s.combinedUsageMax {
 		leastRecentlyModifiedFile := s.getLeastRecentlyModifiedFile()
 
+		if leastRecentlyModifiedFile == nil {
+			ddLog.Warn("getLeastRecentlyModifiedFile returned nil.")
```
[the remark this answers is not in the document]
Comment (soberpeach) [reply pkg/logs/launchers/integration/launcher.go:390]: Change

File: pkg/logs/launchers/integration/launcher.go [hunk 4]
```diff
@@ -67,10 +68,17 @@ func NewLauncher(sources *sources.LogSources, integrationsLogsComp integrations.
...
+	if logsUsageRatio > 1 || logsUsageRatio < 0 {
+		ddLog.Warn("Logs usage ratio setting must be between 0 and 1, current value is:", logsUsageRatio, ". Falling back to integrations_logs_total_usage setting.")
 		maxDiskUsage = logsTotalUsageSetting
+	} else {
+		maxDiskUsage, err = computeMaxDiskUsage(runPath, logsTotalUsageSetting, logsUsageRatio)
+		if err != nil {
+			ddLog.Warn("Unable to compute integrations logs max disk usage, falling back to integrations_logs_total_usage setting:", err)
+			maxDiskUsage = logsTotalUsageSetting
+		}
 	}
```
[the remark this answers is not in the document]
Comment (soberpeach) [reply pkg/logs/launchers/integration/launcher.go:83]: Good idea, added 👍

File: pkg/logs/launchers/integration/launcher_test.go (hunk 1 above)
Comment (soberpeach) [reply pkg/logs/launchers/integration/launcher_test.go:162]: Good idea, adding tests for this now

**Review by gh123man — APPROVED** [review]

**Pull request merged**